src/metadata.jl

```diff
@@ -0,0 +1,124 @@
+import ProtocolBuffers
+
+# Conversion to Julia primitive types
+# ========================================================================================
+const proto_to_julia_type_map = Dict(
+    Protocol_PB.PrimitiveType.INT_128 => Int128,
+    Protocol_PB.PrimitiveType.INT_64 => Int64,
+    Protocol_PB.PrimitiveType.INT_32 => Int32,
+    Protocol_PB.PrimitiveType.INT_16 => Int16,
+    Protocol_PB.PrimitiveType.INT_8 => Int8,
+    Protocol_PB.PrimitiveType.UINT_128 => UInt128,
+    Protocol_PB.PrimitiveType.UINT_64 => UInt64,
+    Protocol_PB.PrimitiveType.UINT_32 => UInt32,
+    Protocol_PB.PrimitiveType.UINT_16 => UInt16,
+    Protocol_PB.PrimitiveType.UINT_8 => UInt8,
+    Protocol_PB.PrimitiveType.FLOAT_64 => Float64,
+    Protocol_PB.PrimitiveType.FLOAT_32 => Float32,
+    Protocol_PB.PrimitiveType.FLOAT_16 => Float16,
+    Protocol_PB.PrimitiveType.STRING => String,
+    Protocol_PB.PrimitiveType.SYMBOL => Symbol,
+    Protocol_PB.PrimitiveType.CHAR => Char,
+    Protocol_PB.PrimitiveType.BOOL => Bool,
+)
```

Comment:
Is there any reason to map to these julia types, rather than just mapping to the types as strings? e.g. 
```julia
    Protocol_PB.PrimitiveType.INT_64 => "Int64",
```
?
Are the types used anywhere other than for printing? I always get nervous when I see types used as values, because it tends to cause both dynamic and compilation heavy code

Reply:
Ah, i see that it's used for `from_proto()`, for constant values, below.. 🤔 Yeah maybe how you have it is right..

I do wonder if it could just be a big switch-statement though. Might be interesting to profile the two cases some day, for schema-heavy queries?

Reply:
> I do wonder if it could just be a big switch-statement though.

Sounds like a good idea to me 👍

Reply:
I leave this as is for now, but add a `TODO` for the future. We could change `_from_proto` to use a big switch statement indeed :)